### ISSUE TITLE
Fix S1481 FN: Support list patterns

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6377]
### Description

FN for S1481 (VariableUnusedBase).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
    public void SomeMethod(object[] byteArray)
    {
        if (byteArray is [1, 2, 3] unusedVar) // FN
        {
        }

        if (byteArray is [SomeClass unusdeVar2, 42]) // FN
        {
        }
    }


```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
